typescript.js

```diff
@@ -1,16 +1,96 @@
 module.exports = {
   parser: '@typescript-eslint/parser',
   plugins: [ '@typescript-eslint/eslint-plugin' ],
-  rules: {
-    'no-extra-parens': 'off',
-    'react/prop-types': 0,
-  },
   overrides: [
     {
       'files': ['**/*.ts', '**/*.tsx'],
+      /**
+       * Many of these rules fall into the category of "Extension Rules", which have the same functionality
+       * as the original rule but with the addition of TypeScript support:
+       * https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin#extension-rules
+       */
       'rules': {
+        'brace-style': 'off',
```

Comment:
I just want to double-check: it was required to override the original ESLint rules for the TS rules to take effect?

In addition, was there a reason for putting the TS rules in `overrides` as opposed to the top-level `rules` config?

Reply:
1. Just using [no-unused-vars](https://github.com/typescript-eslint/typescript-eslint/blob/3e204e9f2fe7008c7d7127d865f58bff1d978e0a/packages/eslint-plugin/docs/rules/no-unused-vars.md) as an example, the extension rules all tell you to disable the base rule.

2. Hypothetically there is no reason. We were previously using overrides for some but not all, so I put them all as overrides for consistency. However, I am happy to defer to the judgement of the stewards if you think they should be top-level instead, since there shouldn't be any reason they couldn't all be top-level as all of the extension rules maintain the functionality of the base rules.

Reply:
1. 🤦 I should have seen that, thanks for pointing out!
2. Gotcha, thanks for the context! I think this is fine, a good benefit I see is co-locating the ESLint rule with the TS rule so it's a bit easier to understand that we are turning 1 off in favor of the other

Reply:
yeah I like the idea of having them together - thanks for adding them this way.